### PR TITLE
Add request to list repositories starred by the authenticated user

### PR DESCRIPTION
**Pull request by kafji**
API doc: https://docs.github.com/en/rest/reference/activity#list-repositories-starred-by-the-authenticated-user

### CODE REVIEW

**Comment by XAMPPRocky** [comment]
Thank you for your PR! This mostly looks good to me, however it should be under `activity` directly and the method should be called `list_repos_starred_by_authenticated_user` to match octokit. https://octokit.github.io/rest.js/v18#activity-list-repos-starred-by-authenticated-user

**Comment by kafji** [comment]
I've moved it to `activity::list_repos_starred_by_authenticated_user` and also renamed the request builder to `ListStarredReposBuilder`.

File: src/api/activity.rs [hunk 1]
```diff
@@ -20,4 +20,83 @@ impl<'octo> ActivityHandler<'octo> {
...
     }
+
+    /// [List repositories starred by current authenticated user](https://docs.github.com/en/rest/reference/activity#list-repositories-starred-by-the-authenticated-user).
+    ///
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// octocrab::instance()
+    ///     .activity()
+    ///     .starring()
+    ///     .list_starred_repositories()
```
Comment (XAMPPRocky) [inline src/api/activity.rs:31]:
```suggestion
    ///     .list_repos_starred_by_authenticated_user()
```

**Comment by kafji** [comment]
I just learned that I don't need to [modify the library for my needs](https://github.com/XAMPPRocky/octocrab#http-api).

Do you think we should keep the PR?

Since the user can add it on their end, I'm thinking we should close it.

**Comment by XAMPPRocky** [comment]
Thank you for your PR! I think we should add it, having a typed API should be preferred if possible.

**Comment by kafji** [comment]
Thank you for merging it.

**Pull request merged**